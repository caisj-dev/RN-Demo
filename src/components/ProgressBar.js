import React from "react";
import { View, Text, Dimensions, StyleSheet } from "react-native";

const { width } = Dimensions.get("window");

const normalise = (value, min, max) =>
  ((value - min) * 100) / (max - min) + "%";

const State = ({ value, max }) => <Text>{value + "/" + max}</Text>;

function ProgressBar({ value, min, max }) {
  return (
    <View
      style={{
        height: 10,
        width: width * 0.5,
        borderRadius: 40,
        backgroundColor: "#f2f2f2",
      }}
    >
      <View
        style={{
          backgroundColor: "#27d2db",
          height: "100%",
          width: `${normalise(value, min, max)}`,
          borderRadius: 40,
        }}
      />
    </View>
  );
}
ProgressBar.State = State;
export default ProgressBar;
