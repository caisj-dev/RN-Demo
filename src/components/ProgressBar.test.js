import React from "react";
import renderer from "react-test-renderer";
import ProgressBar from "./ProgressBar";

describe("<ProgressBar />", () => {
  it("snapshot", () => {
    const tree = renderer
      .create(<ProgressBar value={10} min={0} max={100} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
