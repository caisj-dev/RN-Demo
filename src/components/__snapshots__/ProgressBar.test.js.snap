// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar /> snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f2f2f2",
      "borderRadius": 40,
      "height": 10,
      "width": 375,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#27d2db",
        "borderRadius": 40,
        "height": "100%",
        "width": "10%",
      }
    }
  />
</View>
`;
